### DISCUSSION TITLE
Add the ability to change the background colour of Markdown code blocks

### DISCUSSION THREAD

**Discussion by amarz45** [nvim-treesitter/nvim-treesitter#3835] [General]
I want to be able to easily differentiate between code blocks and regular text in Markdown and R Markdown. I want to do this by changing the backgrond colour of the code blocks. The colour should span across the entire window, not just the background text. Lines with backticks should stay the same colour. I posted this issue on [tree-sitter-markdown](https://github.com/MDeiml/tree-sitter-markdown/issues/64#issuecomment-1279406545) and the owner of that repository told me to ask here instead.

**Comment by ObserverOfTime** [comment]
`after/queries/markdown/highlights.scm`
```query
; extends
(code_fence_content) @text.code
```
`init.lua`
```lua
vim.api.nvim_set_hl(0, '@text.code', {bg = '#000FFF'})
```
---
I don't think it's possible to make it span across the entire window.

**Comment by theHamsta** [comment]
I think it would be possible to set a sign. Their background highlight should be visible for the whole horizontal line. You can set a sign in a directive which can be arbitrary lua code.

**Comment by amarz45** [comment]
@ObserverOfTime thanks for the workaround.

**Comment by clason** [comment]
I'm converting this into a discussion since this is too special to support by default. If someone can share a custom snippet using signs with a directive, that would be appreciated.

**Comment by calind** [comment]
Hi! 

Can you please post how to set the sign for the markdown code block? It would be very helpful as I've been searching for how to do it for a few days.

Thanks!

**Comment by sbatial** [comment]
I don't know if this has been mentioned already but I didn't see it and just found it by coincidence:
`@text.literal` (and `@text.literal.block` if you don't want to have inline code blocks) are built-in (treesitter) highlight groups seem to target exactly that.

I could not find anything else (mis)matching so it should probably be fine.

(Crude) Example with `:hi @text.literal.block guibg=grey`:
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/46973479/2a2dc04b-0966-4acf-b322-c718f09c32cc)
and with `:hi @text.literal guibg=grey`:
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/46973479/e58da6fd-ed8d-4e1b-a5d5-5d4efb1a035d)

<details>
<summary>Relevant settings/plugins for anyone curious:</summary>
<ul>
<li>ft=markdown</li>
<li>markdown plugin: <a href="https://github.com/preservim/vim-markdown">preservim/vim-markdown</a> (with concealing, thus no visible language markers in the pictures)</li>
<li>colorscheme: <a href="https://github.com/EdenEast/nightfox.nvim">dayfox (from EdenEast/nightfox.nvim)</a></li>
</details>

Addition:
I found `#dddddd` as a nice background color:
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/46973479/d92a6d3e-ca4e-4353-a430-6ae617f27fe3)
